src/test/java/org/opensearch/knn/integ/BinaryIndexIT.java

```diff
@@ -0,0 +1,242 @@
+        List<KNNResult> results = parseSearchResponse(EntityUtils.toString(searchResponse.getEntity()), fieldName);
+
+        // Validate
+        assertEquals(k, results.size());
+        for (int i = 0; i < k; i++) {
+            assertEquals(k - i, Integer.parseInt(results.get(i).getDocId()));
+        }
+    }
+
+    @SneakyThrows
+    public void testFaissHnswBinary_when1000Data_thenRecallIsAboveNinePointZero() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        createKnnHnswBinaryIndex(true, KNNEngine.FAISS, indexName, fieldName, 128, null);
+        ingestTestData(indexName, fieldName);
+
+        int k = 100;
+        for (int i = 0; i < testData.queries.length; i++) {
+            List<KNNResult> knnResults = runKnnQuery(indexName, fieldName, testData.queries[i], k);
+            float recall = getRecall(
+                Set.of(Arrays.copyOf(testData.groundTruthValues[i], k)),
+                knnResults.stream().map(KNNResult::getDocId).collect(Collectors.toSet())
+            );
+            assertTrue("Recall: " + recall, recall > 0.9);
+        }
+    }
+
+    @SneakyThrows
+    public void testLuceneHnswBinary_whenKnnIndexIsTrue_thenCreateThrowException() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        Exception e = expectThrows(
+            ResponseException.class,
+            () -> createKnnHnswBinaryIndex(true, KNNEngine.LUCENE, indexName, fieldName, 16, null)
+        );
+        assertTrue(e.getMessage(), e.getMessage().contains("only supported for [faiss] engine"));
+    }
+
+    @SneakyThrows
+    public void testNmslibHnswBinary_whenKnnIndexIsTrue_thenCreateThrowException() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        Exception e = expectThrows(
+            ResponseException.class,
+            () -> createKnnHnswBinaryIndex(true, KNNEngine.NMSLIB, indexName, fieldName, 16, null)
+        );
+        assertTrue(e.getMessage(), e.getMessage().contains("only supported for [faiss] engine"));
+    }
+
+    @SneakyThrows
+    public void testFaissHnswBinary_whenEncodedWithSQ_thenThrowException() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        Exception e = expectThrows(
+            Exception.class,
+            () -> createKnnHnswBinaryIndex(true, KNNEngine.FAISS, indexName, fieldName, 16, ENCODER_SQ)
+        );
+        assertTrue(e.getMessage(), e.getMessage().contains("does not support sq encoder"));
+    }
+
+    @SneakyThrows
+    public void testFaissHnswBinary_whenRadialSearch_thenThrowException() {
```

Comment:
nit: For all throwException tests, a parametrized test is much easier to maintain 

https://github.com/opensearch-project/k-NN/blob/main/src/test/java/org/opensearch/knn/index/query/InvalidSearchQueryIT.java

Reply:
By the way, where is this being used? https://github.com/opensearch-project/k-NN/blob/main/src/test/java/org/opensearch/knn/index/query/InvalidSearchQueryIT.java#L37

Reply:
Got it. I missed the method as the method name does not match with what it actually does..

Reply:
I updatedUsed parameterized test. However, now I remember on reason that I don't like about the parameterized test. When one of test fail, there is no easy way to run that specific test to debug without code change.